### PR TITLE
Add CUDA Ampere configuration support

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: Makefile.kokkos [hunk 1]
```diff
@@ -327,35 +327,27 @@ KOKKOS_INTERNAL_USE_ARCH_PASCAL60 := $(call kokkos_has_string,$(KOKKOS_ARCH),Pas
...
-                                                + $(KOKKOS_INTERNAL_USE_ARCH_PASCAL61)  \
-                                                + $(KOKKOS_INTERNAL_USE_ARCH_PASCAL60)  \
-												+ $(KOKKOS_INTERNAL_USE_ARCH_VOLTA70) \
-												+ $(KOKKOS_INTERNAL_USE_ARCH_VOLTA72) \
-												+ $(KOKKOS_INTERNAL_USE_ARCH_TURING75) \
-                                                + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL50) \
-                                                + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL52) \
-                                                + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL53))
+  KOKKOS_INTERNAL_USE_ARCH_NVIDIA := $(shell expr $(KOKKOS_INTERNAL_USE_ARCH_KEPLER35)  \
+                                                + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL50))
```
Comment (dalg24) [inline Makefile.kokkos:350]: What the deal with that change?

File: cmake/kokkos_arch.cmake [hunk 2]
```diff
@@ -79,6 +80,10 @@ IF (KOKKOS_ENABLE_CUDA)
   IF (KOKKOS_ARCH_VOLTA70 OR KOKKOS_ARCH_VOLTA72)
     SET(KOKKOS_ARCH_VOLTA ON)
   ENDIF()
+
+  IF (KOKKOS_ARCH_AMPERE80)
+	  SET(KOKKOS_ARCH_AMPERE ON)
+  ENDIF()
```
Comment (dalg24) [inline cmake/kokkos_arch.cmake:86]: two-space indentation

File: Makefile.kokkos [hunk 3]
```diff
@@ -327,35 +327,27 @@ KOKKOS_INTERNAL_USE_ARCH_PASCAL60 := $(call kokkos_has_string,$(KOKKOS_ARCH),Pas
...
                                               + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL50) \
                                               + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL52) \
-                                              + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL53))
+                                              + $(KOKKOS_INTERNAL_USE_ARCH_MAXWELL53) \
+                                              + $(KOKKOS_INTERNAL_USE_ARCH_PASCAL61)  \
+                                              + $(KOKKOS_INTERNAL_USE_ARCH_PASCAL60)  \
+                                              + $(KOKKOS_INTERNAL_USE_ARCH_VOLTA70)   \
+                                              + $(KOKKOS_INTERNAL_USE_ARCH_VOLTA72)   \
+                                              + $(KOKKOS_INTERNAL_USE_ARCH_TURING75)  \
+					      + $(KOKKOS_INTERNAL_USE_ARCH_AMPERE80))
```
Comment (masterleinad) [inline Makefile.kokkos:343]:
```suggestion
                                              + $(KOKKOS_INTERNAL_USE_ARCH_AMPERE80))
```

File: cmake/kokkos_arch.cmake [hunk 4]
```diff
@@ -79,6 +80,10 @@ IF (KOKKOS_ENABLE_CUDA)
   IF (KOKKOS_ARCH_VOLTA70 OR KOKKOS_ARCH_VOLTA72)
     SET(KOKKOS_ARCH_VOLTA ON)
   ENDIF()
+
+  IF (KOKKOS_ARCH_AMPERE80)
+	  SET(KOKKOS_ARCH_AMPERE ON)
```
Comment (masterleinad) [inline cmake/kokkos_arch.cmake:85]:
```suggestion
    SET(KOKKOS_ARCH_AMPERE ON)
```

File: Makefile.kokkos (hunk 1 above)
Comment (crtrott) [reply Makefile.kokkos:350]: the conditional was redundant. You only enter this if none of the above were set, and the only ones to check for at this pointer were general Maxwell/Kepler which set Maxwell50 and Kepler35 respectively. So you only need to some up those two at this point.

File: Makefile.kokkos (hunk 3 above)
Comment (crtrott) [reply Makefile.kokkos]: there were some tabs. fixed.

File: cmake/kokkos_arch.cmake (hunk 4 above)
Comment (crtrott) [reply cmake/kokkos_arch.cmake]: fixed

**Comment by crtrott** [comment]
Added the cuda_compute_capability.cc option.

**Pull request merged**